### PR TITLE
Integrate

### PR DESCRIPTION
**Pull request by tgouss**
i don't know how match casadi function and "t" for return x_interpolate and t_interpolate

### CODE REVIEW

File: biorbd_optim/plot.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import multiprocessing as mp
 import numpy as np
 import tkinter
+import casadi
 
-from scipy import interpolate
+from scipy import interpolate, integrate
 from matplotlib import pyplot as plt
```
Comment (pariterre) [inline biorbd_optim/plot.py:7]: It seems that integration was removed from this file. Is this still necessary?

File: biorbd_optim/plot.py [hunk 2]
```diff
@@ -204,52 +205,18 @@ def graphs(self):
...
-            print("Install BiorbdViz if you want to have a live view of the optimization")
+            raise RuntimeError("BiorbdViz must be install to animate the model")
 
-        for i, x_phase in enumerate(x):
-            x_interpolate = np.ndarray((self.ocp.nlp[i]["nbQ"], nb_frames))
-            t_interpolate = np.linspace(0, t[i][-1], nb_frames)
-            for j in range(self.ocp.nlp[i]["nbQ"]):
-                tck = interpolate.splrep(t[i], x_phase[j], s=0)
-                x_interpolate[j] = interpolate.splev(t_interpolate, tck)
+        x = ProblemType.get_states_integrated_from_V(self.ocp, self.sol["x"])
```
Comment (pariterre) [inline biorbd_optim/plot.py:214]: If the function is get_states, I think it should returns the state as is, if the function then it should output the mapped version

File: biorbd_optim/plot.py [hunk 3]
```diff
@@ -204,52 +205,18 @@ def graphs(self):
...
 
+        for i, x_phase in enumerate(x):
             b = BiorbdViz(loaded_model=self.ocp.nlp[i]["model"], **kwargs)
-            b.load_movement(x_interpolate.T)
+            b.load_movement(x_phase.T)
             b.exec()
```
Comment (pariterre) [inline biorbd_optim/plot.py:218]: If ProblemType.get_states_integrated_from_V then I think the mapping should be done here. If we have a method called 
ProblemType.get_q_integrated_from_V, then I would do the mapping inside the function. Do you think it makes sense as an endpoint user?

File: biorbd_optim/problem_type.py [hunk 4]
```diff
@@ -208,14 +208,64 @@ def get_data_from_V(ocp, V, num_phase=None):
         else:
             return q, q_dot, tau
 
+    @staticmethod
+    def get_states_integrated_from_V(ocp, V, number_elements=1, concatenate_phases=False):
+        v = ProblemType.get_data_from_V(ocp, V)[:3]
+
+        if ocp.nb_phases == 1:
+            v = [v]
```
Comment (pariterre) [inline biorbd_optim/problem_type.py:216]: It feels that problem_type is a sketchy place for this method... Is OCP not better?

File: examples/multiphase_torque_driven_ocp/eocar.py [hunk 5]
```diff
@@ -36,10 +36,11 @@ def prepare_ocp(biorbd_model_path="eocar.bioMod", show_online_optim=False, ode_s
     # Constraints
     constraints = (
         (
-            {"type": Constraint.ALIGN_MARKERS, "instant": Instant.START, "first_marker": 0, "second_marker": 1,},
-            {"type": Constraint.ALIGN_MARKERS, "instant": Instant.END, "first_marker": 0, "second_marker": 2,},
+            {"type": Constraint.ALIGN_MARKERS, "instant": Instant.START, "first_marker_idx": 0,
+             "second_marker_idx": 1, },
+            {"type": Constraint.ALIGN_MARKERS, "instant": Instant.END, "first_marker_idx": 0, "second_marker_idx": 2, },
```
Comment (pariterre) [inline examples/multiphase_torque_driven_ocp/eocar.py:41]: Make sure you update with Master. This file doesn't exist anymore

File: examples/multiphase_torque_driven_ocp/load_eocar.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import importlib.util
+from pathlib import Path
+
+from biorbd_optim import OptimalControlProgram, ShowResult
```
Comment (pariterre) [inline examples/multiphase_torque_driven_ocp/load_eocar.py:4]: Can you change the name of the file for save_and_load_something?

File: biorbd_optim/problem_type.py [hunk 7]
```diff
@@ -208,14 +208,64 @@ def get_data_from_V(ocp, V, num_phase=None):
...
+            if same_dof:
+                t_concat = t[0]
+                state_concat = integrated_state[0]
+                for i in range(1, self.ocp.nb_phases):
+                    state_concat = np.concatenate((state_concat, integrated_state[i][:, 1:]), axis=1)
+                    t_concat = np.concatenate((t_concat, t[0][i][1:] + t[-1]))
+        return integrated_state
+
+    @staticmethod
+    def __interpolate(self, idx_phase, x_phase, t, nb_frames):
```
Comment (pariterre) [inline biorbd_optim/problem_type.py:253]: `self` in a staticmethod?

File: biorbd_optim/plot.py (hunk 1 above)
Comment (tgouss) [reply biorbd_optim/plot.py:7]: no !

**Comment by pariterre** [comment]
LGTM

**Pull request merged**